### ISSUE TITLE
Cost tracer log errors

### ISSUE THREAD

**Issue by alfwro13** [BottlecapDave/HomeAssistant-OctopusEnergy#945]
### Describe the bug

I was just going thought my log files troubleshooting some other issue and I can across this error 

`ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved (None)
Traceback (most recent call last):
  File "/config/custom_components/octopus_energy/cost_tracker/cost_tracker.py", line 173, in _async_calculate_cost
    consumption_data = add_consumption(current,
                       ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/octopus_energy/cost_tracker/__init__.py", line 47, in add_consumption
    diff_percentage = abs((diff / mean) * 100)
                           ~~~~~^~~~~~
ZeroDivisionError: float division by zero`

That was logged over 1K in one day. As far I can tell the cost tracker is tracking the cost correctly so it can probably be ignored but clearly there is a problem somewhere

### Reproduction steps

enable cost tracker

### Expected behaviour

n/a

### Tariff Code

E-1R-AGILE-FLEX-22-11-25-J

### Integration Version

11.2.1

### Home Assistant Version

2024.7.1

### Fresh Install?

After upgrading

### Home Assistant Logs

listed above

### Confirmation

- [X] I confirm that I cannot find my solution within the [documentation](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy)
- [X] I confirm that I cannot find my solution within the [FAQ](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq)

**Comment by BottlecapDave** [comment]
Hello and sorry that you're receiving this. I'm currently testing the fix, which should be released within the next couple of days.

**Comment by BottlecapDave** [comment]
This should be fixed in v11.2.2.
